test/config-luatest/basic_test.lua

```diff
@@ -569,3 +570,48 @@ g.test_loader_paths = function(g)
         t.assert_equals(bar.whoami, 'bar')
     end)
 end
+
+-- Verify, that instance can recover from the xlogs using config.
+g.test_recovery_without_uuid = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local cfg = cbuilder.new()
+        :add_instance('instance-001', {
+            database = {
+                mode = 'rw',
+                instance_uuid = '22222222-2222-2222-0022-222222222222',
+            }
+        })
+        :config()
+    local cfg_file = treegen.write_script(dir, 'config.yaml', yaml.encode(cfg))
+    local opts = {
+        config_file = cfg_file,
+        chdir = dir,
+        net_box_credentials = {
+            user = 'client',
+            password = 'secret',
+        },
+    }
+    g.server = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+
+    -- Initial start.
+    g.server:start()
+    g.server:exec(function()
+        t.assert_equals(box.info.status, 'running')
+    end)
+
+    -- Remove UUID. Previously generated ones were used, so it wasn't possible
+    -- to recover from the xlogs without passing UUID to config.
+    local cfg_rs = cfg.groups['group-001'].replicasets['replicaset-001']
+    cfg_rs.instances['instance-001'].database.instance_uuid = nil
+    cfg_file = treegen.write_script(dir, 'config.yaml', yaml.encode(cfg))
+    g.server.config_file = cfg_file
+    -- Recovery process.
+    g.server:restart()
+    g.server:exec(function()
+        t.assert_equals(box.info.status, 'running')
+    end)
+
+    -- Clean everything.
+    g.server:drop()
```

Comment:
It's strange, that all other tests doesn't clean `g` after themselves

Reply:
Also, I don't really know, where to put that test. It's pretty basic)

Reply:
> It's strange, that all other tests here doesn't clean `g` after themselves 

It is in the `after_each` hook, which is set by the `helpers.group()` call.

Such cleaning shouldn't be performed by test cases itself, because if the test case fails, it passes over the clean up. The hook doesn't.

Reply:
> Also, I don't really know, where to put that test. It's pretty basic)

This `basic_test.lua` test is already pretty heavy and I'm a bit nervious about test-run's test timeout.

Personally I just pick up some 'main topic' of a change and if there is no such test, create it. Here we verify a step from the upgrade scenario, so `names_upgrade_test.lua` fits good.

Reply:
I see, it's in `config-luatest/helpers.lua`. Will remove cleaning